documentation/internal/Level-1-Review-And-LOQ.md

```diff
@@ -0,0 +1,120 @@
+# Level 1 Review and LOQ
+
+The aim of this page is to outline the structure and flow of:
+
+- Reviewer - reviewing an application
+- Reviewer - submitting LOQ (list of questions or change request)
+- Applicant - making changes in response to LOQ
+- Applicant - re-submitting application with changes
+- Reviewer - re-reviewing changes that were made
+
+In the context of this document, we are assuming that reviewer is already assigned, all sections and they are the only level reviewer defined for a given stage.
+
+## Brief Summary
+
+Below is the summary of logic aspects of this topic
+
+### Back End
+
+- Changes status of review and application to inform front end of actions that can be made
+- Trims Responses that are duplicated and not changed
+
+### Front End
+
+- Calculates actions and progress of review and application
+- GraphQL mutation to create review and application responses
+- GraphQL mutation to update review_responses
+- Validates submission of review and application
+- GraphQL mutation to request submission of review and application
+
+### Structure
+
+The 'state' of application and review (actions, progress, etc..) is calculated and stored in one state structure. It aids visual component and de-couples logic from UI elements. This structure is recalculated whenever a response is changed
```

Comment:
That's very good, I just missed a little more explanation around what hooks runs in what case, would we require a little more around that? For example, does the `useGetApplicationStructure` re-runs when there is an update to `review_response`?

Reply:
I've added a little bit more: d887cf212f7301f79931fee9c9a53db6409fbc3b, didn't quite add about review_response only affecting only useGetReviewStructureForSection only.  Was hoping that diagram would describe it better:

![Screen Shot 2021-05-21 at 12 22 08 AM](https://user-images.githubusercontent.com/26194949/118977747-9dfbf680-b9ca-11eb-8c46-22836fc7763b.png)